countfiles/utils/file_handlers.py

```diff
@@ -96,3 +99,65 @@ def get_files_without_extension(path: str, recursive=False, include_hidden=True)
         else:
             return [f for f in Path(path).iterdir()
                     if f.is_file() and not is_hidden(f) and not has_extension(f)]
+
+
+def is_hidden_file_or_dir(platform_name: str, path: str) -> bool:
+    """The function determines whether the file or folder is hidden.
+
+    Windows: testing the FILE_ATTRIBUTE_HIDDEN for file or folder
+    (source: https://stackoverflow.com/questions/284115/cross-platform-hidden-file-detection)
+    Linux: testing at least for the dot character in path on Unix-like systems
+    Note: for Linux def is_hidden_file_or_dir('~/path/.to/file.txt') checking for the dot character in path,
+    so if '/.' in path the entire folder is ignored, even if it has visible files.
+    For Windows def is_hidden_file_or_dir('~/path/to/file.txt')
+    checking only file. For checking folder - def is_hidden_file_or_dir('~/path/to_folder')
+    :param platform_name: result of a 'sys.platform' parameter call
+    :param path: ~/path/to/file.txt or ~/path/to_folder
+    :return: True if hidden or False if not
+    """
+    if platform_name.startswith('win'):
+        return bool(os.stat(path).st_file_attributes & stat.FILE_ATTRIBUTE_HIDDEN)
+    elif platform_name.startswith('linux'):
+        return bool('\.' in path or '/.' in path)
```

Comment:
Maybe we should clean the `path` variable  with `os.path.expanduser()`, as all the slashes would then be normalised and the code would be more readable (that way, we only will need to check one condition here: `return '/.' in path`) 

It could also be a good idea to change the `path` name to something that does not conflict with existing modules or packages names, for instance, `filepath`.

Reply:
It's already comes from main.py:
location = os.path.expanduser(args.path)
os.path.normpath() added
Maybe it's useful, maybe not :)